pkg/test/upgrade/upgrade.go

```diff
@@ -0,0 +1,71 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package upgrade
+
+import (
+	"archive/tar"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"istio.io/istio/pkg/test/env"
+)
+
+// Version is the version of the control plane to read installation config for
+type Version string
+
+const (
+	// NMinusOne should be kept at the tip of the previous minor version
+	NMinusOne Version = "1.8.0"
```

Comment:
@Monkeyanator let's step back for a second to think about this for a second ...

I'd like to shift our thinking regarding the way we write tests toward more of a "conformance"-style. By conformance testing, I mean that our tests could theoretically be packaged up and can run against any pre-installed Istio control plane setup. This means that test framework would inherently be decoupled from any implementation of Istio and would likely reside in its own github repository (this has been a long-term goal for a while now). The implication here is that the test framework would no longer be able to install or customize a control plane in anyway. It would simply deploy workloads and configuration and run the tests. Any customizations (certain features enabled/disabled) would require a separate test execution with a different control plane pre-configured.

So that's the vision, and we're not there yet. However, I'd like to make sure that we don't add more dependencies on our Istio implementation/versions which might make decoupling the test framework later more difficult.

With all that said, I think what I'd like to see WRT upgrade testing is something like:

1. Separate the Istio component into 2 parts:
   1. Istio Installer: responsible for installing exactly one version of Istio across all the clusters. This can be run as a setup step. Eventually, I'd like us to stop installing Istio altogether and instead have it installed by the prow scripts before running the tests. Having this separated out into a separate component will simplify that cleanup.
   1. Istio: provides API surface for interacting with all installed control planes which may have been pre-installed or installed by the Istio Installer. 
1. When deploying echo, have the framework pick a control plane version and configure the deployment to use it. This means that the test code will not be aware of the fact that multiple control plane revisions are even being tested.

In addition, the notion of N, N-1, N-1 goes away here as well, since "N" doesn't really mean anything. Instead, we should just configure everything with a set of Istio versions (e.g. 1.7, 1.8, 1.9). Again, the tests should never see this, so they shouldn't care much.